plugins/SlicerT/SlicerTView.cpp

```diff
@@ -43,57 +49,63 @@ namespace lmms {
 namespace gui {
 
 SlicerTView::SlicerTView(SlicerT* instrument, QWidget* parent)
-	: InstrumentViewFixedSize(instrument, parent)
+	: InstrumentView(instrument, parent)
 	, m_slicerTParent(instrument)
+	, m_fullLogo(PLUGIN_NAME::getIconPixmap("full_logo"))
+	, m_background(PLUGIN_NAME::getIconPixmap("toolbox"))
 {
 	// window settings
 	setAcceptDrops(true);
 	setAutoFillBackground(true);
 
-	// render background
-	QPalette pal;
-	pal.setBrush(backgroundRole(), PLUGIN_NAME::getIconPixmap("artwork"));
-	setPalette(pal);
+	setMaximumSize(QSize(10000, 10000));
+	setMinimumSize(QSize(516, 400));
+	setResizable(true);
 
 	m_wf = new SlicerTWaveform(248, 128, instrument, this);
-	m_wf->move(2, 6);
+	m_wf->move(0, s_topBarHeight);
```

Comment:
Michael did point this out, basically, you can use a QVBoxLayout to get the elements listed vertically, and the bottom toolbox can be wrapped in a QHBoxLayout. That way, you'll have to do little calculation.